exporter/signalfxexporter/metricdata_to_signalfxv2.go

```diff
@@ -0,0 +1,403 @@
+		if metric == nil || metric.MetricDescriptor == nil {
+			logger.Warn("Received nil metrics data or nil descriptor for metrics")
+			numDroppedTimeSeries += len(metric.GetTimeseries())
+			continue
+		}
+
+		// Build the fixed parts for this metrics from the descriptor.
+		descriptor := metric.MetricDescriptor
+		metricName := &descriptor.Name
+
+		metricType := fromOCMetricDescriptorToMetricType(descriptor.Type)
+		numLabels := len(descriptor.LabelKeys)
+		filteredLabelKeys := make([]*string, numLabels)
+		for i := 0; i < numLabels; i++ {
+			key := filterKeyChars(descriptor.LabelKeys[i].Key)
+			filteredLabelKeys[i] = &key
+		}
+
+		for _, series := range metric.Timeseries {
+			dimensions := make([]*sfxpb.Dimension, numLabels+numExtraDimensions)
+			copy(dimensions, extraDimensions)
+			for i := 0; i < numLabels; i++ {
+				dimension := &sfxpb.Dimension{
+					Key:   filteredLabelKeys[i],
+					Value: &series.LabelValues[i].Value,
+				}
+				dimensions[numExtraDimensions+i] = dimension
+			}
+
+			for _, dp := range series.Points {
+
+				var msec int64
+				if dp.Timestamp != nil {
+					msec = dp.Timestamp.Seconds*1e3 + int64(dp.Timestamp.Nanos)/1e6
+				}
+
+				sfxDataPoint := &sfxpb.DataPoint{
+					// Source field is not set by code seen at
+					// github.com/signalfx/golib/sfxclient/httpsink.go
+					Metric:     metricName,
+					MetricType: metricType,
+					Timestamp:  &msec,
+					Dimensions: dimensions,
+				}
+
+				switch pv := dp.Value.(type) {
+				case *metricspb.Point_Int64Value:
+					sfxDataPoint.Value = &sfxpb.Datum{IntValue: &pv.Int64Value}
+					sfxDataPoints = append(sfxDataPoints, sfxDataPoint)
+
+				case *metricspb.Point_DoubleValue:
+					sfxDataPoint.Value = &sfxpb.Datum{DoubleValue: &pv.DoubleValue}
+					sfxDataPoints = append(sfxDataPoints, sfxDataPoint)
+
+				case *metricspb.Point_DistributionValue:
+					sfxDataPoints, err = appendDistributionValues(
+						sfxDataPoints,
+						sfxDataPoint,
+						pv.DistributionValue)
+					if err != nil {
+						numDroppedTimeSeries++
+						logger.Warn(
+							"Timeseries for distribution metric dropped",
+							zap.Error(err),
+							zap.String("Metric", *sfxDataPoint.Metric))
+					}
+				case *metricspb.Point_SummaryValue:
+					sfxDataPoints, err = appendSummaryValues(
+						sfxDataPoints,
+						sfxDataPoint,
+						pv.SummaryValue)
+					if err != nil {
+						numDroppedTimeSeries++
+						logger.Warn(
+							"Timeseries for summary metric dropped",
+							zap.Error(err),
+							zap.String("Metric", *sfxDataPoint.Metric))
+					}
+				default:
+					numDroppedTimeSeries++
+					logger.Warn(
+						"Timeseries dropped to unexpected metric type",
+						zap.String("Metric", *sfxDataPoint.Metric))
+				}
+
+			}
+		}
+	}
+
+	return sfxDataPoints, numDroppedTimeSeries, nil
+}
+
+func appendAttributesToDimensions(
+	dimensions []*sfxpb.Dimension,
+	attribs map[string]string,
+) []*sfxpb.Dimension {
+
+	if len(attribs) == 0 {
```

Comment:
This condition is unnecessary since `range attribs` will be a noop and you will just return `dimensions` as it at the end of the function.